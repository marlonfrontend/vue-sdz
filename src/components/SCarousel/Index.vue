<template>
  <coerousel class="s-carousel" v-bind="$attrs">
    <s-icon slot="previous" class="previous" icon="sdz-chevron-left" size="15" />

    <coerousel-item v-for="(item, index) in items" :key="index">
      <slot :item="item" />
    </coerousel-item>

    <s-icon slot="next" class="next" icon="sdz-chevron-right" size="15" />
  </coerousel>
</template>

<script>
import 'vue-coerousel/dist/vue-coerousel.css'
import { Carousel, CarouselItem } from 'vue-coerousel'
// import { Carousel, CarouselItem } from 'vue-coerousel/dist/vue-coerousel.js'

import SIcon from '../SIcon/Index.vue'

export default {
  name: 'SCarousel',

  components: {
    SIcon,
    'coerousel': Carousel,
    'coerousel-item': CarouselItem
  },

  props: {
    items: {
      type: Array,
      required: true
    }
  }
}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.s-carousel {
  & > .controller {
    cursor: pointer;
    border-color: color(primary, base);
    background-color: color(neutral, base);

    & > .icon { color: color(primary, base); }

    & > .previous {
      position: absolute;
      top: 49%;
      left: 13px;
      transform: translateY(-50%);
      font-size: 20px !important;
    }

    & > .next {
      position: absolute;
      top: 49%;
      right: 13px;
      transform: translateY(-50%);
      font-size: 20px !important;
    }
  }

  & > .pagination > .-active { background: color(primary, base); }

  & > .previous { left: -60px; }
  & > .next { right: -60px; }

  &.--is-first-page > .previous { display: none; }
  &.--is-last-page > .next { display: none; }
}
</style>
