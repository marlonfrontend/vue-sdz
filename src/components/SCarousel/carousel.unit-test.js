import { mount, shallowMount } from '@vue/test-utils'

import SCarousel from './Index.vue'

/* eslint-disable */
describe('SCarousel', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(SCarousel, {
      propsData: {
        items: [
          { color: 'blue' },
          { color: 'yellow' },
          { color: 'red' },
          { color: 'black' },
          { color: 'pink' },
          { color: 'turquoise' },
          { color: 'magenta' },
          { color: 'white' },
          { color: 'orange' },
          { color: 'green' },
          { color: 'purple' },
          { color: 'cyan' },
          { color: 'lime' },
          { color: 'lightblue' }
        ],
        breakpoints: {
          1024: { perPage: 4 },
          768: { perPage: 3 },
          640: { perPage: 2 },
          320: { perPage: 1 }
        }
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check items', () => {

    const items = cmp.findAll('.carousel-item')
    expect(items.length).toBe(14);

  })

  test('check class', () => {

    expect(cmp.classes('vue-coerousel')).toBe(true)
    expect(cmp.classes()).toContain('vue-coerousel')

  })


})
