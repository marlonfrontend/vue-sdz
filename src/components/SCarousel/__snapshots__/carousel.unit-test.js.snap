// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCarousel has the expected html structure 1`] = `
<div
  breakpoints="[object Object]"
  class="vue-coerousel s-carousel"
>
  <div
    class="wrapper"
  >
    <div
      class="inner"
      style="transform: translateX(0%);"
    >
       
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
      <div
        class="carousel-item"
        style="flex: 1 0 100%;"
      />
       
    </div>
  </div>
  <div
    class="pagination"
    style="--width: 230px;"
  >
    <div
      class="option -active"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
    <div
      class="option"
    />
  </div>
</div>
`;
